### PR TITLE
Script for testing 2 roles on the first 10 questions in GSM8K

### PR DESCRIPTION
**Pull request by kkahadze**

### CODE REVIEW

File: src/prompt_systematic_review/role_prompting.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+from load_hf_data import load_hf_dataset
+from openai import OpenAI
+from typing import List
+import re
+
+
+def query_model(
+    key: str, prompt: str, question: str, model_name: str, output_tokens: int = 150
+) -> str:
```
Comment (trigaten) [inline src/prompt_systematic_review/role_prompting.py:9]: need documentation throughout file

**Comment by trigaten** [comment]
Resolve merge conflicts and upload another file, in `scripts` that I can run to test the prompts against GSM8K.

File: src/prompt_systematic_review/role_prompting.py (hunk 1 above)
Comment (kkahadze) [reply src/prompt_systematic_review/role_prompting.py:9]: Added

File: src/prompt_systematic_review/role_prompting.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            {"role": "system", "content": prompt},
+            {"role": "user", "content": question},
+        ],
+        max_tokens=output_tokens,
+    )
+
+    return response.choices[0]
+
+
+def evaluate_response(response: str, correct_answer: str) -> bool:
```
Comment (trigaten) [inline src/prompt_systematic_review/role_prompting.py:36]: write a test for this in `tests/`

File: src/prompt_systematic_review/role_prompting.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        if examples and i >= examples:
+            break
+
+    for prompt, result in results.items():
+        accuracy = result["correct"] / result["total"]
+        print(f"Prompt: {prompt}\nAccuracy: {accuracy:.2f}\n")
+
+
+def extract_numbers(string: str) -> List[int]:
+    """
```
Comment (trigaten) [inline src/prompt_systematic_review/role_prompting.py:93]: Write a test for this in `tests`

File: src/prompt_systematic_review/role_prompting.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
+from prompt_systematic_review.load_hf_data import load_hf_dataset
+import openai
+from typing import List
+import re
+
+
+def query_model(
```
Comment (trigaten) [inline src/prompt_systematic_review/role_prompting.py:7]: add a test for this function

File: src/prompt_systematic_review/role_prompting.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    :param correct_answer: The correct answer to the question taken from the dataset.
+    :return: Whether the response is correct.
+    """
+
+    final_answer = extract_numbers(response.message.content)[0]
+    correct = extract_numbers(correct_answer)[0]
+    return final_answer == correct
+
+
+def evaluate_prompts(
```
Comment (trigaten) [inline src/prompt_systematic_review/role_prompting.py:46]: add test for function

**Pull request closed without merging**